Realm 관련 구현

albireo3754 opened the pull request:
# 작업 내용
- Realm API 구현 (CRUD, Search)
- 충돌제거

# 관련된 이슈 번호
closes #19 #27

albireo3754 committed:
✨ 임시로 struct 로 생성한 Entities를 RealmSwift에 사용할 수 있도록 변경

albireo3754 committed:
✨ Realm에 CRUD, Search 추가

albireo3754 committed:
🔧 충돌을 유발하는 gitattributes, gitignore 제거

soohyeon0487 commented on JipJung/JipJung/Domain/Entities/Music.swift:
```diff
@@ -6,12 +6,24 @@
+        super.init()
+    }
+    
+    convenience init(id: String, name: String, mode: Int, thumbnailImageURL: String, videoURL: String, soundURL: String) {
+        self.init()
+        self.id = id
+        self.name = name
+        self.mode = mode
+    }
```
파라미터로 받는 URL 관련 3개의 String은 쓰이지 않는건가요?

soohyeon0487 commented on JipJung/JipJung/Infra/LocalDB/RealmDBManager.swift:
```diff
@@ -0,0 +1,68 @@
+import RealmSwift
+
+protocol LocalDBManageable {
+    func search<T: Object>(with predicate: NSPredicate?) -> Single<Results<T>>?
```
`Single<T>`의 경우, `T` 혹은 `failure(throwable)`이 넘어가는데 별도로 Result Enum을 통해 넘기는 이유가 있나요?

albireo3754 replied:
Results는 success or fail인 기본 Swift타입이 아니라 RealmSwift에서 제공해주는 Array-Like Collection인데 충분히 혼동을 줄 수 있겠다는 생각이드네요.

그래서 Array형식으로 넘겨줄 수 있도록 수정하였습니다.

soohyeon0487 replied:
아니면 Results를 typealias 해서 표기하는 것도 괜찮을 것 같습니다.

soohyeon0487 reviewed (commented):
merge union 관련이 해결되었으므로 gitattrubutes 관련 커밋은 제외해도 되지않을까 싶습니다.

hyunsik-0h commented on JipJung/JipJung/Infra/LocalDB/RealmDBManager.swift:
```diff
@@ -0,0 +1,68 @@
+    func delete(_ value: Object) -> Single<Bool>
+}
+
+class RealmDBManager: LocalDBManageable {
```
찾아보니 이런 것도 있는 것 같습니다.
https://github.com/RxSwiftCommunity/RxRealm

albireo3754 replied:
없는게없네요 ㄷㄷ

albireo3754 committed:
✨ Music 생성자에 빠진내용 추가

albireo3754 committed:
✨ .gitattributes 관련 문제 해결방식 변경에 따른 재 추가

.gitattributes를 사용하지 않기로 논의 했지만, 다시 사용해도 충돌이 없도록 코드를 짜기로 하여 다시 추가하였습니다.

albireo3754 committed:
✨ RealmDBManager search함수 변경

Results가 RealmSwift에서 제공하는 Array형식인데 직관적이지 못해 Swift 기본 타입인 Results와 오해를 불러일으켜서 수정하였습니다.

soohyeon0487 reviewed (approved):
수정 사항 확인하였습니다.
